src/Kernel/IQSharpEngine.cs

```diff
@@ -342,7 +342,31 @@ public override async Task<ExecutionResult> ExecuteMundane(string input, IChanne
                 }
                 catch (CompilationErrorsException c)
                 {
-                    foreach (var m in c.Errors) channel.Stderr(m);
+                    // Check if the user likely tried to execute a magic
+                    // command and try to give a more helpful message in that
+                    // case.
+                    if (input.TrimStart().StartsWith("%") && input.Split("\n").Length == 1)
```

Comment:
since magics accept inputs on multiple lines, I'd say that we shouldn't restrict it to one-line inputs

Reply:
Agreed, but I did it this way to reduce the chance of a false positive (that is, an actual Q# compilation failure gets wrongly treated as a missing magic command). If possible, I'd like to keep it this way for now to be conservative with respect to false positives, and to revisit a more robust way of offering that feedback for multiline magics.